checker/src/org/checkerframework/checker/units/UnitsAnnotatedTypeFactory.java

```diff
@@ -48,7 +48,7 @@
 
     // Map from canonical class name to the corresponding UnitsRelations instance.
     // We use the string to prevent instantiating the UnitsRelations multiple times.
-    private Map<String, UnitsRelations> unitsRel;
+    private static Map<String, UnitsRelations> unitsRel;
```

Comment:
Don't make things static. Each checker/atf should manage their state in instance fields.